src/opm/output/eclipse/Tables.cpp

```diff
@@ -1245,6 +1323,77 @@ namespace { namespace SatFunc {
             });
         }
 
+        /// Create linearised and padded 'TAB' vector entries of normalised
+        /// SWFN tables for all saturation function regions from Family One
+        /// table data (SGWFN keyword).
+        ///
+        /// \param[in] numRows Number of rows to allocate for each table in
+        ///    the output vector.  Expected to be equal to the number of
+        ///    declared saturation nodes in the simulation run's TABDIMS
+        ///    keyword (Item 3).
+        ///
+        /// \param tolcrit Minimum relative permeability threshold value for
+        ///    phase to be considered mobile.  Values less than this threshold
+        ///    are output as zero.
+        ///
+        /// \param[in] swof Collection of SWOF tables for all saturation
+        ///    regions.
+        ///
+        /// \return Linearised and padded 'TAB' vector values for output
+        ///    SWFN tables.  Corresponds to unit-converted copies of columns
+        ///    1, 3, and 4 --with added derivatives--of the input SGWFN tables.
+        std::vector<double>
+        fromSGWFN(const std::size_t          numRows,
+                  const double               tolcrit,
+                  const Opm::TableContainer& swfn)
+        {
+            using SWFN = ::Opm::SgwfnTable;
+
+            const auto numTab = swfn.size();
+            const auto numDep = std::size_t{2}; // Krw, Pcow
+
+            return detail::createSatfuncTable(numTab, numRows, numDep,
+                [tolcrit, &swfn]
+                    (const std::size_t           tableID,
+                     const std::size_t           primID,
+                     Opm::LinearisedOutputTable& linTable)
+                -> std::size_t
+            {
+                const auto& t = swfn.getTable<SWFN>(tableID);
+
+                auto numActRows = std::size_t{0};
+
+                // Sw
+                {
+                    const auto& Sw = t.getSgColumn();
+
+                    numActRows = Sw.size();
+                    std::copy(std::begin(Sw), std::end(Sw),
+                              linTable.column(tableID, primID, 0));
+                }
+
+                // Krgw(Sw)
+                detail::outputRelperm(t.getKrgwColumn(), tolcrit,
+                                      linTable.column(tableID, primID, 1));
```

Comment:
I think you need to do more than this.  The first column in the result array is supposed to be `Sw` in increasing order and you can't just copy `Sg` into that and use it directly.  The second column is supposed to be `krgw` in increasing order.  You'll have to use something along the lines of
```C++
{
    const auto& Sg = t.getSgColumn();
    std::transform(std::rbegin(Sg), std::rend(Sg),
                   linTable.column(tableID, primID, 0),
                   [](const auto sg) { return 1.0 - sg; });
}

{
    const auto& krgw = t.getKrgwColumn();
    detail::outputRelperm(std::rbegin(krgw), std::rend(krgw), tolcrit,
                          linTable.column(tableID, primID, 1));
}
````
instead.